### PR TITLE
Refactor: enable flexible external configuration

### PR DESCRIPTION
**Pull request by sergeyfrolov**
Most deletions are due to removal of Android test traces.

### CODE REVIEW

File: u_common.go [hunk 1]
```diff
@@ -74,49 +74,68 @@ const (
...
-	HelloAndroid_Auto        ClientHelloID = ClientHelloID{helloAndroid, helloAutoVers}
-	HelloAndroid_6_0_Browser ClientHelloID = ClientHelloID{helloAndroid, 23}
-	HelloAndroid_5_1_Browser ClientHelloID = ClientHelloID{helloAndroid, 22}
+	HelloChrome_Auto = HelloChrome_62
+	HelloChrome_58   = ClientHelloID{helloChrome, 58}
+	HelloChrome_62   = ClientHelloID{helloChrome, 62}
 )
 
+// based on spec's GreaseStyle, GREASE_PLACEHOLDER may be replaced by another GREASE value
+const GREASE_PLACEHOLDER = 0x0a0a
```
Comment (willscott) [inline u_common.go]: is this gonna have to be updated every 6 weeks or whatever?

Comment (sergeyfrolov) [reply u_common.go]: why would it have to be updated every 6 weeks or whatever?

Comment (willscott) [reply u_common.go]: I see, this is the template pattern that stays stable. makes sense.

File: handshake_messages.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ type clientHelloMsg struct {
 	serverName                   string
 	ocspStapling                 bool
 	scts                         bool
-	ems                          bool
+	ems                          bool // [UTLS]
```
Comment (bemasc) [inline handshake_messages.go]: This comment could use some expansion.

File: u_common.go [hunk 3]
```diff
@@ -74,49 +74,68 @@ const (
 	helloRandomizedNoALPN
 )
 
+type ClientHelloSpec struct {
+	ClientRandom       []byte
```
Comment (bemasc) [inline u_common.go]: Can this be nil?

File: u_common.go [hunk 4]
```diff
@@ -74,49 +74,68 @@ const (
...
+	CipherSuites       []uint16       // nil => default
+	CompressionMethods []uint8        // nil => no compression
+	Extensions         []TLSExtension // nil => no extensions
+
+	// GreaseStyle: currently only Chrome(GREASE depends on ClientRandom)
```
Comment (bemasc) [inline u_common.go]: Does GREASE still depend on ClientRandom?

File: handshake_messages.go (hunk 2 above)
Comment (sergeyfrolov) [reply handshake_messages.go]: Throughout the standard files, which we are not supposed to change too much, I was leaving `[UTLS]` comment to explicitly state what we added.
I can switch to something like `[Added by UTLS]`, if that's better

File: u_common.go (hunk 3 above)
Comment (sergeyfrolov) [reply u_common.go]: Yes, but now that I thought of it, we probably shouldn't have it in spec at all, as it's supposed to be a generic descriptor of configuration.
Will remove.

File: u_common.go (hunk 4 above)
Comment (sergeyfrolov) [reply u_common.go]: Good point, forgot to update this.

File: handshake_messages.go (hunk 2 above)
Comment (bemasc) [reply handshake_messages.go]: I understood that it was marked as added.  I was hoping you would expand the comment to explain _why_ it was added.

Comment (sergeyfrolov) [reply handshake_messages.go]: Alright, `[UTLS]` -> `[UTLS] extension implemented due to its prevalence`

**Pull request merged**